Lib/ldapurl.py

```diff
@@ -130,58 +132,64 @@ def __ne__(self,other):
     return not self.__eq__(other)
 
 
-class LDAPUrlExtensions(UserDict):
-  """
-  Models a collection of LDAP URL extensions as
-  dictionary type
-  """
-
-  def __init__(self,default=None):
-    UserDict.__init__(self)
-    for k,v in (default or {}).items():
-      self[k]=v
-
-  def __setitem__(self,name,value):
+class LDAPUrlExtensions(MutableMapping):
     """
-    value
-        Either LDAPUrlExtension instance, (critical,exvalue)
-        or string'ed exvalue
+    Models a collection of LDAP URL extensions as
+    a mapping type
     """
-    assert isinstance(value,LDAPUrlExtension)
-    assert name==value.extype
-    self.data[name] = value
 
-  def values(self):
-    return [
-      self[k]
-      for k in self.keys()
-    ]
-
-  def __str__(self):
-    return ','.join(str(v) for v in self.values())
-
-  def __repr__(self):
-    return '<%s.%s instance at %s: %s>' % (
-      self.__class__.__module__,
-      self.__class__.__name__,
-      hex(id(self)),
-      self.data
-    )
+    def __init__(self, default=None):
+        self._data = {}
+        if default is not None:
+            self.update(default)
+
+    def __setitem__(self, name, value):
+        """
+        value
+            Either LDAPUrlExtension instance, (critical,exvalue)
+            or string'ed exvalue
+        """
+        assert isinstance(value, LDAPUrlExtension)
+        assert name == value.extype
```

Comment:
While you are fixing the issue, please replace assert with proper checks. asserts are dangerous because they are ignored with ``python -O``.